docker/docker-compose.yml

```diff
@@ -0,0 +1,34 @@
+version: '3.7'
+services:
+  workout-lol:
+    build: 
+      context: ..
+      dockerfile: docker/dockerfile
+    ports:
+      - 3000:3000
+    depends_on:
+      mongodb:
+        condition: service_healthy
+
+  mongodb:
+    image: mongo:7.0.0-rc6
+    environment:
+      - MONGO_INITDB_ROOT_USERNAME=mongoadmin
+      - MONGO_INITDB_ROOT_PASSWORD=secret
+      - MONGO_INITDB_DATABASE=db
+    volumes:
+      - ./mongo-data:/data/db
+    healthcheck:
+      test: echo 'db.stats().ok' | mongosh localhost:27017/test --quiet
+      interval: 30s
+      timeout: 10s
+      retries: 5
+      start_period: 30s
+
+  mongoseed:
```

Comment:
Should we attempt to do the seeding within the main app itself ... in an idempotent way? I've observed that various self-hosted apps (like Firefly-III, Bookstack etc) follow that principle for ease of setup.

That is, making use of the `Workout` APIs instead of connecting to MongoDB directly.

Reply:
I think that both ways are ok.
I prefer using a separate container on small projects as it is easier to debug
It also permit not to install mongo binaries on the app image (lighter image).
But it's maybe easier to just run everything in a single container.